### PR TITLE
Patching development branch to 1.7.2 game version

### PR DESCRIPTION
**Pull request by WakooMan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] All new classes have class-level documentation comments, if there are any at all
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Now the mod works with 1.7.2 game version
- [x] Most likely imports/usings changed in the mod (about 75%).
- [x] Some serializers needed to be changed to work correctly.
- [x] Added some suggestions in comment for removing empty cases in serializers in one of the serializer file.
- [x] Town Sync is almost done, but not completely, so it's commented.

## What is the current behaviour?

The mod works with 1.7.1 game version.

## What is the new behaviour?

The mod works with 1.7.2 game version.
Resolves #

## Other information

### CODE REVIEW

**Comment by garrettluskey** [comment]
Town sync should be separate from the 1.7.2 update. You can base a "TownSync" branch off of this one to work with 1.7.2, but since these are different features they should be in separate branches.

File: source/Coop/Mod/GameSync/Initializer.cs [hunk 1]
```diff
@@ -29,6 +31,11 @@ public static void SetupSyncAfterLoad()
                 FiefSync.MakeManaged(town);
             }
 
+            //foreach (Town town in Town.AllTowns)
```
Comment (garrettluskey) [inline source/Coop/Mod/GameSync/Initializer.cs]: Move this commented code to a different branch

File: source/Coop/Mod/GameSync/Town/TownCommands.cs [hunk 2]
```diff
@@ -0,0 +1,296 @@
+﻿//using System;
```
Comment (garrettluskey) [inline source/Coop/Mod/GameSync/Town/TownCommands.cs]: Move this code to a different branch

File: source/Coop/Mod/GameSync/Town/TownSync.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
+﻿//using CoopFramework;
```
Comment (garrettluskey) [inline source/Coop/Mod/GameSync/Town/TownSync.cs]: Move this code to a separate branch

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/Coop/Mod/GameSync/Initializer.cs (hunk 1 above)
Comment (WakooMan) [reply source/Coop/Mod/GameSync/Initializer.cs]: Done

File: source/Coop/Mod/GameSync/Town/TownCommands.cs (hunk 2 above)
Comment (WakooMan) [reply source/Coop/Mod/GameSync/Town/TownCommands.cs]: Done

File: source/Coop/Mod/GameSync/Town/TownSync.cs (hunk 3 above)
Comment (WakooMan) [reply source/Coop/Mod/GameSync/Town/TownSync.cs]: Done

**Review by garrettluskey — APPROVED** [review]

**Pull request merged**